dev/WebView2/WebView2AutomationPeer.cpp

```diff
@@ -0,0 +1,258 @@
+}
+
+winrt::AutomationControlType WebView2AutomationPeer::GetAutomationControlTypeCore()
+{
+    return winrt::AutomationControlType::Pane;
+}
+
+#if WINUI3
+HRESULT WebView2AutomationPeer::GetRawElementProviderSimple(_Outptr_opt_ IRawElementProviderSimple** value)
+{
+    InitProvider();
+    m_provider.copy_to(value);
+    return S_OK;
+}
+
+HRESULT WebView2AutomationPeer::IsCorrectPeerForHwnd(HWND hwnd, _Out_ bool* value)
+{
+    *value = false;
+    if (!InitProvider())
+    {
+        return S_OK;
+    }
+
+    auto hwndProvider = GetImpl()->GetProviderForHwnd(hwnd).try_as<IRawElementProviderSimple>();
+    if (hwndProvider && hwndProvider == m_provider)
+    {
+        *value = true;
+    }
+    return S_OK;
+}
+#endif
+
+struct ProviderWrapper : winrt::implements<ProviderWrapper, winrt::IInspectable, ::IRawElementProviderSimple, ::IRawElementProviderSimple2, ::IRawElementProviderFragment, ::IRawElementProviderFragmentRoot>
+{
+public:
+    ProviderWrapper(winrt::com_ptr<IRawElementProviderSimple> const& child, winrt::FrameworkElementAutomationPeer parent) : m_child(child), m_parent(parent)
+    {
+    }
+
+    HRESULT STDMETHODCALLTYPE get_ProviderOptions(
+        __RPC__out enum ProviderOptions* pRetVal) override
+    {
+        *pRetVal = ProviderOptions_ServerSideProvider | ProviderOptions_UseComThreading
+            | ProviderOptions_OverrideProvider;
+
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetPatternProvider(
+        PATTERNID patternId,
+        __RPC__deref_out_opt IUnknown** pRetVal) override
+    {
+        *pRetVal = nullptr;
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetPropertyValue(
+        PROPERTYID propertyId,
+        __RPC__out VARIANT* pRetVal) override
+    {
+        if (propertyId == UIA_IsControlElementPropertyId) {
+            pRetVal->vt = VT_BOOL;
+            pRetVal->boolVal = VARIANT_TRUE;
+        }
+        else if (propertyId == UIA_IsContentElementPropertyId) {
+            pRetVal->vt = VT_BOOL;
+            pRetVal->boolVal = VARIANT_TRUE;
+        }
+        else {
+            return E_NOTIMPL;
+        }
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE get_HostRawElementProvider(
+        __RPC__deref_out_opt IRawElementProviderSimple** pRetVal) override
+    {
+        return m_child->get_HostRawElementProvider(pRetVal);
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE ShowContextMenu(void) override
+    {
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE Navigate(
+        /* [in] */ enum NavigateDirection direction,
+        /* [retval][out] */ __RPC__deref_out_opt IRawElementProviderFragment** pRetVal) override
+    {
+        if (direction == NavigateDirection_Parent)
+        {
+            *pRetVal = m_parent.as<::IRawElementProviderFragment>().detach();
+        }
+        else
+        {
+            *pRetVal = nullptr;
+        }
+        return S_OK;
+    }
+
+    HRESULT STDMETHODCALLTYPE GetRuntimeId(
```

Comment:
Should all these ABI method calls have a `try { } CATCH_RETURN()` around them? Minimally, the `check_hresult` in this method will generate a C++ exception which ABIs don't typically like.

Reply:
noexcept will put a try/catch -> std::terminate, which is preferred since any exception from here is our fault. I notice that noexcept isn't on the other methods so I will add it.